File: Striver Important Interview Questions/Day 15 - String/reverse-words-in-a-string.cpp

// https://leetcode.com/problems/reverse-words-in-a-string/

class Solution {
public:
    string reverseWords(string s) {

        string ans = "";
        if(s.size() == 0) return ans;

        stringstream ss(s);
        string temp = "";
        vector<string> words;
        while(ss >> temp) words.push_back(temp);

        for(int i = words.size()-1; i >= 0; i--){
            if(i == 0) ans += words[i];
            else ans += words[i] + " ";
        }

        return ans;

    }
};

File: Striver A2Z/Step 5 - String [Basic and Medium]/Lec 2 - Medium String Problems/reverse-words-in-a-string.js

// https://leetcode.com/problems/reverse-words-in-a-string/

/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(str) {
    return str.trim().split(" ").filter(ele => ele !== "").reverse().join(" ");
};
